sematic/db/queries.py

```diff
@@ -305,6 +305,60 @@ def save_job(job: Job) -> Job:
         return job
 
 
+def get_runs_with_orphaned_jobs() -> List[str]:
+    with db().get_session() as session:
+        query_results = list(
+            session.query(Job.run_id, Job.kind, Job.state, Run.id, Run.future_state)
+            .filter(Job.run_id == Run.id)
+            .filter(Job.kind == JobKind.run)
+            .filter(
+                Run.future_state.in_(
+                    [state.value for state in FutureState.terminal_states()]
+                )
+            )
+            .filter(Job.state.not_in(KubernetesJobState.terminal_states()))
+            .all()
+        )
+        run_ids = set()
+        for _, __, job_state, run_id, future_state in query_results:
+            logger.info(
+                "Run %s in state %s has orphaned job in state %s",
+                run_id,
+                future_state,
+                job_state,
+            )
+            run_ids.add(run_id)
```

Comment:
It will be "lossy" in jobs that get logged to console this way (since you can in principle have multiple orphaned jobs per run id), but it is possible to express. Changed.